src/guide/bash.md

```diff
@@ -241,14 +241,48 @@ With a domain available, it is time to register an account.
 
 ## 4. Registering an Account
 
-To register a new account within the _looking_glass_ domain, run:
+To register a new account called _mad_hatter_ within the _looking_glass_
+domain, run:
 
 ```bash
 ./iroha_client_cli account register \
     --id="mad_hatter@looking_glass" \
     --key="ed0120a753146e75b910ae5e2994dc8adea9e7d87e5d53024cfa310ce992f17106f92c"
 ```
 
+The `--id` argument in the above code snippet specifies the _account id_, the unique name assigned to that account in the blockchain, which includes the _domain_ which is a group of things like asset definitions, account id's and other objects that we'll talk about later. The `--key` argument specifies the
+[public key](https://en.wikipedia.org/wiki/Public-key_cryptography), think of it as the "password" used by `mad_hatter@looking_glass`. 
+
+Keys come always come in pairs. It is possible to generate a pair of `public` and `private` keys using one of our
+tools, `kagami`. When we've released Iroha, you should be able to install it separately and call like every other program on your system, but for now you can either ask `cargo` to `run` it for you, or build the program and find it in one of the target directories (either `target/debug` or `target/release`). 
+
+To generate a new key pair, run it without any arguments.
+
+
+To use `cargo`, run 
+```bash
+$ cargo run --bin kagami  -- crypto
+Public key (multihash): ed012088a5d0cada6edfd4ad8046da64ffce820c16f0e68a21f98df505becd5011e18a
+Private key: 749682bd40695cfc77b5ad599220a80d69b72db5e5078c259635f8e8d8f3559088a5d0cada6edfd4ad8046da64ffce820c16f0e68a21f98df505becd5011e18a
+Digest function: ed25519
+```
+
+::: info 
+You can also use the `-a` flag to specify the algorithm  for key-pair generation and the `-s`
+flag to specify the seed:
+
+For example, to generate an `ed25519` key (the default algorithm). 
```

Comment:
I'd use another (not the default) algorithm as an example here. You can generate the default without the `-a`, it'd make more sense to illustrate the use of this flag with another algorithm

Reply:
Thanks, while I tried adding an example for `secp256k1`, I've found an LTS bug, so I won't be marking this section as resolved yet.